.github/actions/generate-attestations/src/main.ts

```diff
@@ -0,0 +1,58 @@
+import * as core from "@actions/core";
+import fs from "fs";
+import path from "path";
+import { writeAttestations } from "./attestation";
+
+// Detect directory traversal for input file.
+function resolvePathInput(input: string, wd: string): string {
+  const safeJoin = path.join(wd, input);
+  if (!safeJoin.startsWith(wd)) {
```

Comment:
I think you need to normalize the wd to have a trailing `/` before calling startsWith`, otherwise:
`wd: /some/path`
`input: ../path-other/afile`
`safeJoin = /some/path-other/afile` would pass the verification

Reply:
adding tests, since this wasn't clear to me